### ISSUE TITLE
Refactor Menu Constraints, Dependencies and Validations

### ISSUE THREAD

**Issue by gregorydimitriadis** [DCS-LCSR/SignStream3#579]
Currently the code used in menus to control option selection is adhoc.  The way constraints, dependencies and validations of options is for the most part all one offs.  This is prone to errors.

**Comment by douglas-motto-at-rutgers** [comment]
This is a catch all issue for systematic problems with menus.  Several problems include...

i) Menu options are not constrained to prevent invalid selections.

ii) Menu options with invalid selection may be "Continue"d, "OK"ed, "Enter"ed, "Save"d, etc..And therefore saved as such.

iii) Menu objects do not close cleanly, and when open do not create fresh...resulting in options selected from initially opened menus from appearing selected in follow up opened menus...even when the two menu opens represent completely independent items.

iv) Menus which open other sub menus aren't consistently or reliably controlled as to when multiple menus may be allowed to edited at the same time, or only the most child menu is editable at one time (requiring its closure to allow parents to be editable again).

**Comment by douglas-motto-at-rutgers** [comment]
I do not believe this issue involves problems the navigational menu items as well, but it may.
